admin/metrics/utils.py

```diff
@@ -0,0 +1,93 @@
+"""
+Metrics scripts
+"""
+from datetime import datetime, timedelta
+from modularodm import Q
+from website.project.model import User, Node
+
+from .models import OSFWebsiteStatistics
+
+DAY_LEEWAY = 86400 - 60  # How many seconds are counted as a full day for the
+# last day
+
+
+def get_list_of_dates(start, end):
+    delta = end - start
+    if delta.seconds > DAY_LEEWAY:
+        delta = delta + timedelta(hours=1)
+    return [start + (s * timedelta(days=1)) for s in xrange(1, delta.days + 1)]
+
+
+def get_previous_midnight(time=None):
+    if time is None:
+        time = datetime.utcnow()
+    return time - timedelta(  # As close to midnight utc as possible
+        hours=time.hour,
+        minutes=time.minute,
+        seconds=time.second,
+        microseconds=time.microsecond - 1
+    )
+
+
+def get_osf_statistics(time=None):
+    """ get all dates since the latest
+
+    :param time: immediately turns into the previous midnight
+    :return: nothing
+    """
+    time = get_previous_midnight(time)
+    latest = None
+    if OSFWebsiteStatistics.objects.count() != 0:
+        latest = OSFWebsiteStatistics.objects.latest('date')
+        if latest.date.date() == time.date():
+            return  # Don't add another
+        dates = get_list_of_dates(latest.date, time)
+    else:
+        dates = [time]
+    for date in dates:
+        get_days_statistics(date, latest)
+        latest = OSFWebsiteStatistics.objects.latest('date')
+
+
+def get_days_statistics(time, latest=None):
+    statistic = OSFWebsiteStatistics(date=time)
```

Comment:
`statistic` &rarr; `statistics` to keep up with 8aa95a5d3125cee13d0e67f36251aa12d97b5187.